nixos/modules/services/misc/paperless.nix

```diff
@@ -0,0 +1,194 @@
+      description = "Whether all users can write to the consumption dir.";
+    };
+
+    ocrLanguages = mkOption {
+      type = with types; nullOr (listOf string);
+      default = null;
+      description = ''
+        Languages available for OCR via Tesseract, specified as
+        <literal>ISO 639-2/T</literal> language codes.
+        If unset, defaults to all available languages.
+      '';
+      example = [ "eng" "spa" "jpn" ];
+    };
+
+    address = mkOption {
+      type = types.str;
+      default = "localhost";
+      description = "Server listening address.";
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 28981;
+      description = "Server port.";
+    };
+
+    extraConfig = mkOption {
+      type = types.attrs;
+      default = {};
+      description = ''
+        Extra paperless config options.
+
+        The config values are evaluated as double-quoted Bash string literals.
+
+        See <literal>paperless-src/paperless.conf.example</literal> for available options.
+
+        To enable user authentication, set <literal>PAPERLESS_DISABLE_LOGIN = "false"</literal>
+        and run the shell command <literal>$dataDir/paperless-manage createsuperuser</literal>.
+
+        To define secret options without storing them in /nix/store, use the following pattern:
+        <literal>PAPERLESS_PASSPHRASE = "$(&lt; /etc/my_passphrase_file)"</literal>
+      '';
+      example = literalExample ''
+        {
+          PAPERLESS_OCR_LANGUAGE = "deu";
+        }
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = defaultUser;
+      description = "User under which Paperless runs.";
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.paperless;
+      defaultText = "pkgs.paperless";
+      description = "The Paperless package to use.";
+    };
+
+    manage = mkOption {
+      type = types.package;
+      readOnly = true;
+      default = manage;
+      description = ''
+        A script to manage the Paperless instance.
+        It wraps Django's manage.py and is also available at
+        <literal>$dataDir/manage-paperless</literal>
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable (
+    let
+      preStart = ''
+        if [[ ! -e "${cfg.dataDir}" ]]; then
+          install -o ${cfg.user} -g $(id -gn ${cfg.user}) -d "${cfg.dataDir}"
+        fi
+        if [[ $(readlink ${cfg.dataDir}/paperless-manage) != ${manage} ]]; then
+          ln -sf ${manage} ${cfg.dataDir}/paperless-manage
+        fi
+      '' + optionalString cfg.consumptionDirIsPublic ''
+        if [[ ! -e "${cfg.consumptionDir}" ]]; then
+          install -o ${cfg.user} -g $(id -gn ${cfg.user}) -m 777 -d "${cfg.consumptionDir}"
+        fi
+      '' + ''
+        exec ${pkgs.libuuid}/bin/runuser -u "${cfg.user}" -- ${setupDB}
+      '';
+
+      # Auto-migrate on first run or if the package has changed
+      setupDB = pkgs.writeScript "setupDB" ''
```

Comment:
```suggestion
      setupDB = pkgs.writeBash "setupDB" ''
```

Reply:
`writeBash` would imply an additional `set -e` line in the script. I don't see the gain here.

Reply:
Sorry, that was a misfire. Of course, `writeBash` is a better fit. Fixed: https://github.com/NixOS/nixpkgs/pull/44086/commits/c4209b00e36d3e84df108b237f26c46af50acad7